search/example-cloud-function/main.py

```diff
@@ -0,0 +1,212 @@
+      ),
+      spell_correction_spec=discoveryengine.SearchRequest.SpellCorrectionSpec(
+          mode=discoveryengine.SearchRequest.SpellCorrectionSpec.Mode.AUTO
+      ),
+  )
+
+  # Execute the search and return the response as proto.
+  response = client.search(request)
+  # Get the documents as a simple dict for JSON response.
+  documents = [
+      get_document_info(Message.to_dict(result)) for result in response.results
+  ]
+
+  output = {
+      "summary": response.summary.summary_text,
+      "total_size": response.total_size,
+      "attribution_token": response.attribution_token,
+      "next_page_token": response.next_page_token,
+      "documents": documents,
+  }
+  return json.dumps(output)
+
+
+@functions_framework.http
+def vertex_search(request):
+  """HTTP Cloud Function.
+
+  Args:
+      request (flask.Request): The request object.
+        <https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data>
+
+  Returns:
+      The response text, or any set of values that can be turned into a
+      Response object using `make_response`
+      <https://flask.palletsprojects.com/en/1.1.x/api/#flask.make_response>.
+  """
+  # Set CORS headers for the preflight request
+  if request.method == "OPTIONS":
+    # Allows GET requests from any origin with the Content-Type
+    # header and caches preflight response for an 3600s
+    headers = {
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": "POST",
+        "Access-Control-Allow-Headers": "Content-Type",
+        "Access-Control-Max-Age": "3600",
+    }
+
+    return ("", 204, headers)
+
+  # Set CORS headers for the main request
+  headers = {"Access-Control-Allow-Origin": "*"}
+
+  # Handle a variety of inputs types and params.
+  if isinstance(request, str):
+    search_term = request
+  else:
+    if "content-type" not in request.headers:
+      return ("Please specify a 'content-type' header", 422, headers)
+
+    content_type = request.headers["content-type"]
+    if "application/json" in content_type:
+      request_json = request.get_json(silent=True)
+      if request_json and "search_term" in request_json:
+        search_term = request_json["search_term"]
+      else:
+        return (
+            "JSON is invalid, or missing a 'search_term' property",
+            422,
+            headers,
+        )
+    elif "application/octet-stream" in content_type:
+      search_term = request.data
+    elif "text/plain" in content_type:
+      search_term = request.data
+    elif "application/x-www-form-urlencoded" in content_type:
+      search_term = request.form.get("search_term")
+    else:
+      return (f"Unknown content type: {content_type}", 422, headers)
+
+  results = query_vertex_search_api_for_term(search_term)
+
+  # Remove HTML and URL formatting from response.
+  results = re.sub("<.*?>", "", results)
+  results = html.unescape(results)
```

Comment:
nit: There should be some fields in the `DiscoveryEngine.SearchResponse` that already have these values escaped/unescaped. But I'm not sure which are which.